From 8f6adef67c93d04a9eefa1404db0c8de9877fb38 Mon Sep 17 00:00:00 2001
From: sudhanshukumar22 <sudhanshu.kumar@broadcom.com>
Date: Tue, 26 Nov 2024 00:40:46 -0800
Subject: [PATCH] Pull request: https://github.com/FRRouting/frr/pull/17487
 bgpd: Do not reset peers on suppress-fib toggling If the desired state is the
 same - do nothing instead of resetting once again.

Fixes: bdb5ae8 (bgpd: Make suppress-fib-pending clear peering)

Signed-off-by: Donatas Abraitis <donatas@opensourcerouting.org>
---
 bgpd/bgpd.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/bgpd/bgpd.c b/bgpd/bgpd.c
index d6d874be2..0b2bf821e 100644
--- a/bgpd/bgpd.c
+++ b/bgpd/bgpd.c
@@ -461,6 +461,10 @@ void bgp_suppress_fib_pending_set(struct bgp *bgp, bool set)
 	if (bgp->inst_type == BGP_INSTANCE_TYPE_VIEW)
 		return;
 
+	/* Do nothing if already in a desired state */
+	if (set == !!CHECK_FLAG(bgp->flags, BGP_FLAG_SUPPRESS_FIB_PENDING))
+		return;
+
 	if (set) {
 		SET_FLAG(bgp->flags, BGP_FLAG_SUPPRESS_FIB_PENDING);
 		/* Send msg to zebra for the first instance of bgp enabled
-- 
2.39.4

